src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509AuthorityInformationAccessExtension.cs

```diff
@@ -0,0 +1,320 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Formats.Asn1;
+using System.Security.Cryptography.Asn1;
+using System.Security.Cryptography.X509Certificates.Asn1;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    /// <summary>
+    ///   Represents the Authority Information Access X.509 Extension (1.3.6.1.5.5.7.1.1).
+    /// </summary>
+    public sealed class X509AuthorityInformationAccessExtension : X509Extension
+    {
+        private AccessDescriptionAsn[]? _decoded;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="X509AuthorityInformationAccessExtension" />
+        ///   class.
+        /// </summary>
+        public X509AuthorityInformationAccessExtension()
+            : base(Oids.AuthorityInformationAccess)
+        {
+            _decoded = Array.Empty<AccessDescriptionAsn>();
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="X509AuthorityInformationAccessExtension" />
+        ///   class from an encoded representation of the extension and an optional critical marker.
+        /// </summary>
+        /// <param name="rawData">
+        ///   The encoded data used to create the extension.
+        /// </param>
+        /// <param name="critical">
+        ///   <see langword="true" /> if the extension is critical;
+        ///   otherwise, <see langword="false" />.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="rawData" /> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <paramref name="rawData" /> did not decode as an Authority Information Access extension.
+        /// </exception>
+        public X509AuthorityInformationAccessExtension(byte[] rawData, bool critical = false)
+            : base(Oids.AuthorityInformationAccess, rawData, critical)
+        {
+            _decoded = Decode(RawData);
```

Comment:
Does it matter whether this uses rawData or RawData?  It looks like the latter is just a clone of the former.

Reply:
For well behaved apps it doesn't matter. For multi-threaded writes, using the property is more consistent with the delayed reads from, e.g., loading the data in via CopyFrom(AsnEncodedData).